L1Trigger/L1THGCal/plugins/concentrator/HGCalConcentratorProcessorSelection.cc

```diff
@@ -0,0 +1,53 @@
+#include "L1Trigger/L1THGCal/interface/concentrator/HGCalConcentratorProcessorSelection.h"
+#include <limits>
+
+#include "DataFormats/HGCDigi/interface/HGCDigiCollections.h"
+
+DEFINE_EDM_PLUGIN(HGCalConcentratorFactory, 
+        HGCalConcentratorProcessorSelection,
+        "HGCalConcentratorProcessorSelection");
+
+HGCalConcentratorProcessorSelection::HGCalConcentratorProcessorSelection(const edm::ParameterSet& conf)  : 
+  HGCalConcentratorProcessorBase(conf),
+  choice_(conf.getParameter<std::string>("Method")),
+  concentratorProcImpl_(conf)
+{ 
+}
+
+void HGCalConcentratorProcessorSelection::run(const edm::Handle<l1t::HGCalTriggerCellBxCollection>& triggerCellCollInput, 
+                                              l1t::HGCalTriggerCellBxCollection& triggerCellCollOutput,
+                                              const edm::EventSetup& es)
+{ 
+  const l1t::HGCalTriggerCellBxCollection& collInput = *triggerCellCollInput;
+
+  std::unordered_map<uint32_t, std::vector<l1t::HGCalTriggerCell>> tc_modules;
+  for(const auto& trigCell : collInput) {
+    uint32_t module = geometry_->getModuleFromTriggerCell(trigCell.detId());
+    auto itr_insert = tc_modules.emplace(module, std::vector<l1t::HGCalTriggerCell>());
```

Comment:
I think you can simplify these two lines, since you are default-constructing the value:
```
tc_modules[module].push_back(trigCell);
```